### PR TITLE
Hash email

### PR DESCRIPTION
**Pull request by RobStallion**
hash users email addresses #247

### CODE REVIEW

File: priv/repo/migrations/20180508133435_add_email_hash_to_user.exs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    end)
+    |> Enum.map(&Repo.update_all(&1, []))
+  end
+
+  def down do
+    from(u in "users",
+      update: [set: [email: u.email]]
+    )
+    |> Repo.update_all([])
+  end
```
Comment (Danwhy) [inline priv/repo/migrations/20180508133435_add_email_hash_to_user.exs]: This doesn't actually do anything any more does it? It seems a bit weird, but could we just leave this as an empty function?

File: web/controllers/auth.ex [hunk 2]
```diff
@@ -53,7 +54,9 @@ defmodule Bep.Auth do
 
   def login_by_email_and_pass(conn, email, given_pass, opts) do
     repo = Keyword.fetch!(opts, :repo)
-    user = repo.get_by(Bep.User, email: String.downcase(email))
+    repo.all(User)
```
Comment (Danwhy) [inline web/controllers/auth.ex]: Is this line supposed to be here?

File: priv/repo/migrations/20180508133435_add_email_hash_to_user.exs (hunk 1 above)
Comment (RobStallion) [reply priv/repo/migrations/20180508133435_add_email_hash_to_user.exs]: Good spot 👍 🥇. Will update this now thanks

File: web/controllers/auth.ex (hunk 2 above)
Comment (RobStallion) [reply web/controllers/auth.ex]: No 🙈. Thanks for pointing that out.

**Pull request merged**